src/System.IO.Pipelines/tests/PipePoolTests.cs

```diff
@@ -266,6 +267,114 @@ public async Task OnReaderCompletedCalledAfterBlocksReturned()
             pipe.Reader.Complete();
         }
 
+        [Fact]
+        public async Task WritesToArrayPoolByDefault()
+        {
+            var pipe = new Pipe();
+            pipe.Writer.WriteEmpty(10);
+            await pipe.Writer.FlushAsync();
+            pipe.Writer.Complete();
+
+            ReadResult result = await pipe.Reader.ReadAsync();
+            Assert.Equal(10, result.Buffer.Length);
+
+            SequenceMarshal.TryGetReadOnlySequenceSegment(
+               result.Buffer,
+               out ReadOnlySequenceSegment<byte> start,
+               out int startIndex,
+               out ReadOnlySequenceSegment<byte> end,
+               out int endIndex);
+
+            var startSegment = (BufferSegment)start;
+            var endSegment = (BufferSegment)end;
+
+            Assert.Same(startSegment, endSegment);
+            Assert.NotNull(startSegment.Memory);
+            Assert.IsType<byte[]>(startSegment.MemoryOwner);
+
+            pipe.Reader.AdvanceTo(result.Buffer.End);
+            pipe.Reader.Complete();
+        }
+
+        [Fact]
+        public async Task GetMemoryOverMaxPoolSizeAllocatesArray()
+        {
+            using (var pool = new DisposeTrackingBufferPool())
+            {
+                var pipe = new Pipe(new PipeOptions(pool: pool));
+
+                // Allocate 5 KB
+                pipe.Writer.WriteEmpty(5 * 1024);
```

Comment:
Is this testing for `(sizeHint > _pool.MaxBufferSize)`? Is `MaxBufferSize` 4096 for the `MemoryPool` being used? If not, we should add such a test.

Reply:
Yes it is.